cli/src/main/java/com/devonfw/tools/ide/tool/PluginBasedCommandlet.java

```diff
@@ -1,195 +1,233 @@
+   */
+  public void uninstallPlugin(PluginDescriptor plugin) {
+
+    Path pluginsPath = getPluginsInstallationPath();
+    if (!Files.isDirectory(pluginsPath)) {
+      this.context.debug("Omitting to uninstall plugin {} ({}) as plugins folder does not exist at {}",
+          plugin.getName(), plugin.getId(), pluginsPath);
+      return;
+    }
+    FileAccess fileAccess = this.context.getFileAccess();
+    Path match = fileAccess.findFirst(pluginsPath, p -> p.getFileName().toString().startsWith(plugin.getId()), false);
+    if (match == null) {
+      this.context.debug("Omitting to uninstall plugin {} ({}) as plugins folder does not contain a match at {}",
+          plugin.getName(), plugin.getId(), pluginsPath);
+      return;
+    }
+    fileAccess.delete(match);
+  }
+
+  /**
+   * Retrieves the plugin descriptor for the specified key, which can be either the ID or filename of the plugin properties file.
+   *
+   * @param key The key representing the plugin, which can be either the ID or filename
+   * (excluding the ".properties" extension) of the plugin configuration file.
+   * @return the {@link PluginDescriptor} for the given {@code key}.
+   */
+  public PluginDescriptor getPlugin(String key) {
+
+    if (key == null) {
+      return null;
+    }
+    if (key.endsWith(IdeContext.EXT_PROPERTIES)) {
+      key = key.substring(0, key.length() - IdeContext.EXT_PROPERTIES.length());
+    }
+
+    PluginDescriptor pluginDescriptor = getPluginsMap().getById().get(key);
+
+    if (pluginDescriptor == null) {
+      pluginDescriptor = getPluginsMap().getByName().get(key);
+    }
+
+    if (pluginDescriptor == null) {
+      throw new CliException(
+          "Could not find plugin " + key + " at " + getPluginsConfigPath().resolve(key) + ".properties");
+    }
+    return pluginDescriptor;
+  }
+
+  /**
+   * Iterates over plugins from both maps by name and ID, installing missing plugins and handling inactive ones.
+   */
+  @Override
+  protected boolean doInstall(boolean silent) {
+
+    boolean newlyInstalled = super.doInstall(silent);
+    // post installation...
+    boolean installPlugins = newlyInstalled;
+    Path pluginsInstallationPath = getPluginsInstallationPath();
+    if (newlyInstalled) {
+      this.context.getFileAccess().delete(pluginsInstallationPath);
+    } else if (!Files.isDirectory(pluginsInstallationPath)) {
+      installPlugins = true;
+    }
+    if (installPlugins) {
+      Map<String, PluginDescriptor> mapById = getPluginsMap().getById();
+      Map<String, PluginDescriptor> mapByName = getPluginsMap().getByName();
+
+      for (PluginDescriptor plugin : mapByName.values()) {
+        if (!mapById.containsKey(plugin.getId())) {
+          if (plugin.isActive()) {
+            installPlugin(plugin);
+          } else {
+            handleInstall4InactivePlugin(plugin);
+          }
+        }
+      }
+      for (PluginDescriptor plugin : mapById.values()) {
```

Comment:
should IMHO now be simplified and would be sufficient as:
```suggestion
      for (PluginDescriptor plugin : getPluginsMap().getPlugins()) {
          if (plugin.isActive()) {
            installPlugin(plugin);
          } else {
            handleInstall4InactivePlugin(plugin);
          }
        }
      }
```